clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -1434,12 +1445,17 @@ bool FunctionVariableConstraint::solutionEqualTo(Constraints &CS,
 std::string FunctionVariableConstraint::mkString(Constraints &CS,
                                                  bool EmitName, bool ForItype,
                                                  bool EmitPointee,
-                                                 bool UnmaskTypedef) const {
+                                                 bool UnmaskTypedef,
+                                                 std::string UseName) const {
+  if (UseName.empty())
+    UseName = Name;
   std::string Ret = ReturnVar.mkTypeStr(CS, false);
   std::string Itype = ReturnVar.mkItypeStr(CS);
   // This is done to rewrite the typedef of a function proto
   if (UnmaskTypedef && EmitName)
-    Ret += Name;
+    Ret += UseName;
+  else if (EmitName)
+    Ret += "(*" + UseName + ")";
```

Comment:
What's happening here? Maybe add a comment, since I suspect the logic is subtle.

Reply:
And also, maybe it's clearer to write it
```
if (EmitName) {
  if (UnmaskTypedef)
    Ret += UseName;
  else
    Ret += "(*" + UseName + ")";
}
```

Reply:
I'll change it. I really only added to this section what I needed. I didn't follow up to see how the unmasking worked. I usually do, but this seemed a bit too off topic for now.